app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -105,6 +107,40 @@ public void saveBook(View view)
         }
     }
 
+    /**
+     * Validate the fields entered in this activity
+     * Title and author cannot be empty
+     * ISBN can be empty, or has digits or length 10 or 13
+     *
+     * @return true if all fields are valid, false otherwise
+     */
+    private boolean fieldsValid()
+    {
+        boolean valid = true;
+        if (TextUtils.isEmpty(titleView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(titleView);
+            valid = false;
+        }
+        if (TextUtils.isEmpty(authorView.getText().toString()))
+        {
+            EditTextValidator.isEmpty(authorView);
+            valid = false;
+        }
+        // ISBN can be empty, or has digits of length 10 or 13
+        // https://en.wikipedia.org/wiki/International_Standard_Book_Number
+        String isbn_input = isbnView.getText().toString();
```

Comment:
Are we sure we want to allow this to be empty? Later down the line we might be able to compare it to scans to make sure users are handing over the right book

Reply:
We should ask the TA if isbn is mandatory

Reply:
Might as well ask about description too while we're at it